mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/AutoScaleMetricsConfig.java

```diff
@@ -98,6 +120,16 @@ public AggregationAlgo getAggregationAlgo(final String metricGroupName, final St
         return getAllMetrics().keySet();
     }
 
+    public Set<String> generateSourceJobMetricGroups(Set<String> clientIds) {
+        Set<String> results = new HashSet<>();
+        for (String clientId : clientIds) {
+            for (String metricPattern : sourceJobMetrics.keySet()) {
+                results.add(metricPattern.replaceAll(CLIENT_ID_TOKEN, clientId));
```

Comment:
We could probably just look for all ServerSentEventRequestHandler dropped metrics with current-job-Id(should be in the clientId by default?) in the metric group name.. There will also be sockAddr in the metricGroupName, so there will be multiple metric groups per clientId anyways.

Reply:
The job can have multiple clientIds. The patterns generated would be `ServerSentEventRequestHandler:clientId=clientId1:*`, etc. `AutoScaleMetricConfigTest` should make this clear. Is this what you mean?

Reply:
nevermind, I was thinking the `clientId` would always have the current jobID in it, so we could just look for `ServerSentEventRequestHandler:.*clientId=<currentJobId>.*` but it looks like `clientId` can be overridden in the TargetInfo job param.